tests/transfer/test_plugins.py

```diff
@@ -0,0 +1,182 @@
+            self.assertEqual(
+                menu_items[0].url,
+                "/en/admin/cms/page/plugin/plugin_importer/export-plugins/?language=en&placeholder=1"
+            )
+            self.assertEqual(
+                menu_items[1].url,
+                "/en/admin/cms/page/plugin/plugin_importer/import-plugins/?language=en&placeholder=1"
+            )
+
+    def test_import_plugin_views(self):
+        with self.login_user_context(self.user):
+            response = self.client.get(admin_reverse("cms_import_plugins"))
+            self.assertEqual(response.content, b"Form received unexpected values.")
+
+        del self.user # self.get_request() checks for "user" attribute
+        self.assertRaises(
+            PermissionDenied,
+            self.plugin_importer.import_plugins_view,
+            self.get_request()
+        )
+
+    def test_import_plugin_views_for_page(self):
+        with self.login_user_context(self.user):
+            # GET page import
+            response = self.client.get(
+                admin_reverse("cms_import_plugins") + f"?language=en&cms_pagecontent={self.page_content.id}"
+            )
+            self.assertEqual(response.templates[0].name, "djangocms_transfer/import_plugins.html")
+            self.assertEqual(response.context["form"].initial["cms_pagecontent"], self.page_content)
+
+            # POST page import
+            post_data = {
+                "language": "en", "cms_pagecontent": [self.page_content.id],
+                "import_file": SimpleUploadedFile("file.txt", bytes(json.dumps(self._get_expected_page_export_data()), "utf-8"))
+            }
+            response = self.client.post(
+                admin_reverse("cms_import_plugins") + f"?language=en&cms_pagecontent={self.page_content.id}",
+                post_data
+            )
+            self.assertIn(b'<div class="success"></div>', response.content)
+
+    def test_import_plugin_views_for_placeholder(self):
+        placeholder = self.page_content.get_placeholders().get(slot="content")
+        with self.login_user_context(self.user):
+            # GET placeholder import
+            response = self.client.get(
+                admin_reverse("cms_import_plugins") + f"?language=en&placeholder={placeholder.id}"
+            )
+            self.assertEqual(response.templates[0].name, "djangocms_transfer/import_plugins.html")
+            self.assertEqual(response.context["form"].initial["placeholder"], placeholder)
+
+            # empty placeholder import (no existing plugin)
+            request_path = admin_reverse("cms_import_plugins") + f"?language=en&placeholder={placeholder.id}"
+            post_data = {
+                "language": "en", "placeholder": [placeholder.id],
+                "import_file": SimpleUploadedFile("file.txt", b"[null, null]")
+            }
+            with self.assertRaises(IndexError):
+                self.client.post(path=request_path, data=post_data)
```

Comment:
@fsbraun 
I don't this is a plugin-specific test. The error is raised from `get_plugin_tree`
helper method